server/src/main/java/org/opensearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -1521,7 +1521,7 @@ public long getRestoreThrottleTimeInNanos() {
         return restoreRateLimitingTimeInNanos.count();
     }
 
-    protected void assertSnapshotOrGenericThread() {
+    protected void assertAllowableThreadPools() {
```

Comment:
Maybe revert this renaming since the method is otherwise unchanged now? It will make the PR quite a bit cleaner by reducing the number of files touched too.

Reply:
yeah it's now like out of scope of this PR to rename this method but I think it is fine to keep this change as the new method name is more generic ? unless you have a strong opinion about it

Reply:
No strong opinion